asciidoctorj-core/src/main/java/org/asciidoctor/ast/AbstractNodeImpl.java

```diff
@@ -54,157 +66,215 @@ public DocumentRuby document() {
+        return getBoolean("set_attr", name, value, overwrite);
     }
 
     @Override
     public boolean isOption(Object name) {
-        return this.abstractNode.isOption(name);
+        return getBoolean("option?", name);
     }
 
     @Override
     public boolean isRole() {
-        return this.abstractNode.isRole();
+        return getBoolean("role?");
     }
 
     @Override
     public String getRole() {
-        return this.abstractNode.getRole();
+        return getString("role");
     }
 
     @Override
     public String role() {
-        return this.abstractNode.role();
+        return getRole();
     }
 
     @Override
     public List<String> getRoles() {
-        return this.abstractNode.getRoles();
+        return getList("roles", String.class);
     }
 
     @Override
     public boolean hasRole(String role) {
-        return this.abstractNode.hasRole(role);
+        return getBoolean("role?", role);
     }
 
     @Override
     public boolean isReftext() {
-        return this.abstractNode.isReftext();
+        return getBoolean("reftext?");
     }
 
     @Override
     public String getReftext() {
-        return this.abstractNode.getReftext();
+        return getString("reftext");
     }
 
     @Override
     public String iconUri(String name) {
-        return this.abstractNode.iconUri(name);
+        return getString("icon_uri", name);
     }
 
     @Override
     public String mediaUri(String target) {
-        return this.abstractNode.mediaUri(target);
+        return getString("media_uri", target);
     }
 
     @Override
     public String imageUri(String targetImage) {
-        return this.abstractNode.imageUri(targetImage);
+        return getString("image_uri", targetImage);
     }
 
     @Override
     public String imageUri(String targetImage, String assetDirKey) {
-        return this.abstractNode.imageUri(targetImage, assetDirKey);
+        return getString("image_uri", targetImage, assetDirKey);
     }
 
     @Override
     public String readAsset(String path, Map<Object, Object> opts) {
-        return this.abstractNode.readAsset(path, RubyHashUtil.convertMapToRubyHashWithSymbolsIfNecessary(runtime, opts));
+        return getString("read_asset", path, RubyHashUtil.convertMapToRubyHashWithSymbolsIfNecessary(runtime, opts));
     }
 
     @Override
     public String normalizeWebPath(String path, String start, boolean preserveUriTarget) {
-        return this.abstractNode.normalizeWebPath(path, start, preserveUriTarget);
+        return getString("normalize_web_path", path, start, preserveUriTarget);
     }
 
     @Override
     public String getStyle() {
+        return getString("@style");
+    }
 
-        IRubyObject style = ((RubyObjectHolderProxy) this.abstractNode).__ruby_object()
-                .getInstanceVariables()
-                .getInstanceVariable("@style");
+    @Override
+    public String listMarkerKeyword() {
+        return getString("list_marker_keyword");
```

Comment:
We probably want to make this getListMarkerKeyword. Though, honestly, this is an internal method and probably shouldn't be exposed.